src/native/minipal/file.c

```diff
@@ -0,0 +1,198 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <stdint.h>
+#include <minipal/file.h>
+
+#ifdef TARGET_WINDOWS
+#include <Windows.h>
+
+#define ExtendedPrefix L"\\\\?\\"
+#define DevicePathPrefix L"\\\\.\\"
+#define UNCExtendedPathPrefix L"\\\\?\\UNC\\"
+#define UNCPathPrefix L"\\\\"
+#define DIRECTORY_SEPARATOR_CHAR L'\\'
+#define ALT_DIRECTORY_SEPARATOR_CHAR L'/'
+#define VOLUME_SEPARATOR_CHAR L':'
+
+static bool IsDirectorySeparator(WCHAR ch)
+{
+    return (ch == DIRECTORY_SEPARATOR_CHAR) || (ch == ALT_DIRECTORY_SEPARATOR_CHAR);
+}
+
+static bool IsPathNotFullyQualified(WCHAR* path)
```

Comment:
```suggestion
static bool IsPathNotFullyQualified(WCHAR* path, size_t length)
```

Pass in the path length and assert it is sufficient since we are indexing into the array.